pkg/controller/che/k8s_helpers.go

```diff
@@ -269,16 +272,61 @@ func (r *ReconcileChe) GetEndpointTlsCrt(instance *orgv1.CheCluster, url string)
 		// sometimes timing conditions apply, and host isn't available right away
 		if len(testRoute.Spec.Host) < 1 {
 			time.Sleep(time.Duration(1) * time.Second)
-			testRoute := r.GetEffectiveRoute(instance, "test")
-			requestURL = "https://" + testRoute.Spec.Host
+			testRoute = r.GetEffectiveRoute(instance, "test")
 		}
 		requestURL = "https://" + testRoute.Spec.Host
 
 	} else {
-		requestURL = url
+		requestURL = endpointUrl
+	}
+
+	//adding the proxy settings to the Transport object
+	transport := &http.Transport{}
+
+	if instance.Spec.Server.ProxyURL != "" {
+		logrus.Infof("Configuring proxy with %s to extract crt from the following URL: %s", instance.Spec.Server.ProxyURL, requestURL)
+		proxyParts := strings.Split(instance.Spec.Server.ProxyURL, "://")
+		proxyProtocol := ""
+		proxyHost := ""
+		if len(proxyParts) == 1 {
+			proxyProtocol = ""
+			proxyHost = proxyParts[0]
+		} else {
+			proxyProtocol = proxyParts[0]
+			proxyHost = proxyParts[1]
+
+		}
+
+		proxyURL := proxyHost
+		if instance.Spec.Server.ProxyPort != "" {
+			proxyURL = proxyURL + ":" + instance.Spec.Server.ProxyPort
+		}
+		if len(instance.Spec.Server.ProxyUser) > 1 && len(instance.Spec.Server.ProxyPassword) > 1 {
+			proxyURL = instance.Spec.Server.ProxyUser + ":" + instance.Spec.Server.ProxyPassword + "@" + proxyURL
+		}
+
+		if proxyProtocol != "" {
+			proxyURL = proxyProtocol + "://" + proxyURL
+		}
+		config := httpproxy.Config{
+			HTTPProxy:  proxyURL,
+			HTTPSProxy: proxyURL,
+			NoProxy:    strings.Replace(instance.Spec.Server.NonProxyHosts, "|", ",", -1),
+		}
+		proxyFunc := config.ProxyFunc()
+		transport.Proxy = func(r *http.Request) (*url.URL, error) {
+			theProxyUrl, err := proxyFunc(r.URL)
+			if err != nil {
+				logrus.Warnf("Error when trying to get the proxy to access TLS endpoint URL: %s - %s", r.URL, err)
+			}
+			logrus.Infof("Using proxy: %s to access TLS endpoint URL: %s", theProxyUrl, r.URL)
+			return theProxyUrl, err
```

Comment:
AFAIU It would lead to a situation where test route is not removed and can't be created anymore since it exists 

Consider adding fetching route before try to create it to avoid such situation https://github.com/eclipse/che-operator/pull/66/commits/44212ab8136ae5af62c390ac396e9db728ca0356#diff-5899627ff2ee9c6f7afb34d27f2d3b3eR262

Reply:
BTW is the proxy URL returned as certificate by purpose?

Reply:
Not sure I understand your questions here.This is the return of a *proxy function* that is passed as an argument to the transport object. We're not returning from the main function here.

Reply:
Ah, sorry. I need to be more attentive.